distributed/protocol/core.py

```diff
@@ -49,7 +49,9 @@ def dumps(msg, serializers=None, on_error="message", context=None):
             if "lengths" not in head:
                 head["lengths"] = tuple(map(nbytes, frames))
             if "compression" not in head:
-                frames = frame_split_size(frames)
+                # splitting frames is not the default behavior for UCX
+                if split_frames:
+                    frames = frame_split_size(frames)
```

Comment:
It looks like we only consider splitting frames due to compression.  I wonder if we can make this a bit simpler and avoid the keyword argument if we always include `"compression": False` in the header?

Reply:
That would probably work but there may be a GPU compression story in the future.  I suppose we could handle that then if/when it becomes available

Reply:
Give me a couple minutes to look things over

Reply:
I still suggest that we go with setting `compression=False` for now rather than introduce the new keyword argument.  

Tying frame splitting to compression is not entirely clean, but it does happen to be the only reason to split frames today.  In the future if that continues to be the case then we might have compressors register a maximum frame size.  If that doesn't continue to be the case (perhaps some comms also have a maximum size limit) then we will have to rejigger this code regardless.  

However, looking at this a bit more, I notice that `cuda_serialize` already specifies a compression value.  My guess is that this is coming from serializing tuples/lists/dicts of objects, and that we don't move compression values through in that case.  I'll take a look at this later tonight.

Reply:
Yeah ok, so coupling compression with serialization gets complicated, at least with our current logic.  

https://github.com/dask/distributed/blob/511427b81f599105dc1eb4d2f35fd33aa249b4b8/distributed/protocol/core.py#L51-L57

Currently a serializer can choose to pre-compress data.  If so, it adds the compression used to the header.  Great, we can pass on compression (and thus frame splitting).  The CUDA serializers avoids this by setting `compression=None` claiming "I've already handled compression, don't bother" to downstream.  Great.  If we send a single cupy array or something along then we'll be fine.

However, if we happen to include a few things in a small dict/tuple/list then we lose this information.  The headers for the individual bits get thrown into a subheader, and so we lose direct access to the compression.

https://github.com/dask/distributed/blob/511427b81f599105dc1eb4d2f35fd33aa249b4b8/distributed/protocol/serialize.py#L144-L181

Currently our logic though is "If you specify any compression anywhere, don't do anything".  But really we want to handle this on a frame-by-frame basis.  Otherwise the first time someone sends a combined result of `[numpy_array, cupy_array]` things will get messy.  

I think I'm shaving a yak here, but it could be that the right thing to do here is to rework how we handle compression on a frame by frame basis.  Submitting PR now.

Reply:
Well the other thing I was thinking about was checking `head` for `"cuda"` not in `"serializer"`, which would also be easy and avoid this.